### ISSUE TITLE
The CRD generation for SciFi resources needs custom fields' descriptions for references

### ISSUE THREAD

**Issue by ziyue-101** [GoogleCloudPlatform/k8s-config-connector#1931]
### Checklist

- [X] I did not find a related open enhancement request.
- [X] I understand that enhancement requests filed in the GitHub repository are by default low priority.
- [X] If this request is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

### Describe the feature or resource

 The new SciFi workflow  uses controller-gen **https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/dev/tasks/generate-crds#L28** to generate CRDs from go structs(SOT).
 
For some resources like GKEHubFeatureMembership, KCC needs custom field description such as.
```
field: configmanagement.configSync.metricsGcpServiceAccountRef.external
Description: "The Email of the Google Cloud Service Account (GSA) used for exporting Config Sync metrics to Cloud Monitoring. The GSA should have the Monitoring Metric Writer(roles/monitoring.metricWriter) IAM role. The Kubernetes ServiceAccount `default` in the namespace `config-management-monitoring` should be bound to the GSA. Allowed value: The `email` field of an `IAMServiceAccount` resource."
```
However, the existing CRD go types reuses a common ResourceRef [here](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/ea7ee5cdbe1b91f4350dbaff2897419c947983e8/pkg/clients/generated/apis/gkehub/v1beta1/gkehubfeaturemembership_types.go#L49), which makes it hard to customize attributes like field description for individual resources

I can think of two approaches to address the limitation.

* Make each resource has its own resource reference instead of sharing the same reference struct. This allows us to have customizations easily.
* Keep the current shared resource reference. But we need to build some CRD overrides mechanism to support customization and that could scatter attributes for the CRDs in different places.  

I am leaning towards the first option and if the team agrees, we should call out that part in the SciFi howtos.

**Comment by ziyue-101** [comment]
/cc @acpana @justinsb @yuwenma 
WDYT?

**Comment by acpana** [comment]
> Make each resource has its own resource reference instead of sharing the same reference struct. This allows us to have customizations easily.

This may be closer to the final stage in that we use a references definition from a go struct that we control. I think you may be on the first reference out to a GSA account as a direct resource.

**Comment by yuwenma** [comment]
Two suggestions:
1. It's great to explore all possible options and compare the pros and cons. Regarding the question you have, I'm also leaning towards a dedicated resolution for each resource ref. Here's [an example](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1920/files#diff-7d016c9a3cb2e2748a4fba699e47711239a50c5ab21986062a6cdd2eae2eccdd) that may unblock your case:
  -  For projectRef, the PR supports two forms: one from generic resource ref, for backward compatibility with TF generated APIs, the other is a kind-specific reference.
  - For computenetworkRef, the PR gives an example that it re-defines the spec reference and validates/resolves the actual compute network at the beginning of the reconciliation. I think that may suite your case here. 
 
2. We only hit 1-2 resource reference use cases in the direct controller, I would encourage to  build your resource reference that best suite the use case you are dealing with and do not bother to abstract a common pattern (yet).   

BTW, we shall not refer internal codenames or links in oss.

**Comment by ldanielmadariaga** [comment]
This item has been open for >12 months. As part of scheduled repository maintenance, this Issue/PR will be closed. If this is still a critical ask, please re-open and work with your Google Cloud Account team to prioritize it with GCP KCC Product Management.
